File: app/src/main/java/com/djrausch/billtracker/adapters/RepeatingSpinnerAdapter.java
package com.djrausch.billtracker.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.djrausch.billtracker.models.RepeatingItem;

/**
 * Created by white on 7/25/2016.
 */
public class RepeatingSpinnerAdapter extends ArrayAdapter<RepeatingItem> {
    public RepeatingSpinnerAdapter(Context context, int resource) {
        super(context, resource);
        setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view;
        TextView text;

        if (convertView == null) {
            view = LayoutInflater.from(parent.getContext()).inflate(android.R.layout.simple_spinner_item, parent, false);
        } else {
            view = convertView;
        }

        text = (TextView) view;

        RepeatingItem item = getItem(position);
        text.setText(item.title);

        return view;
    }

    @Override
    public TextView getDropDownView(int position, View convertView, ViewGroup parent) {
        View view;
        TextView text;

        if (convertView == null) {
            view = LayoutInflater.from(parent.getContext()).inflate(android.R.layout.simple_spinner_dropdown_item, parent, false);
        } else {
            view = convertView;
        }

        text = (TextView) view;

        RepeatingItem item = getItem(position);
        text.setText(item.title);

        return text;
    }
}

File: app/src/main/java/com/djrausch/billtracker/adapters/RepeatingSpinnerAdapter.kt
package com.djrausch.billtracker.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.djrausch.billtracker.models.RepeatingItem


class RepeatingSpinnerAdapter(context: Context, resource: Int) : ArrayAdapter<RepeatingItem>(context, resource, RepeatingItem.getItems(context)) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view: View
        val text: TextView

        if (convertView == null) {
            val inflater = LayoutInflater.from(parent.context)
            view = inflater.inflate(android.R.layout.simple_spinner_item, parent, false)
        } else {
            view = convertView
        }

        text = view as TextView

        val item = getItem(position)
        text.text = item.title

        return view
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): TextView {
        val view: View
        val text: TextView

        if (convertView == null) {
            val inflater = LayoutInflater.from(parent.context)
            view = inflater.inflate(android.R.layout.simple_spinner_dropdown_item, parent, false)
        } else {
            view = convertView
        }

        text = view as TextView

        val item = getItem(position)
        text.text = item.title

        return text


        /*TextView v = (TextView) super.getView(position, convertView, parent);

        v.setText(getItem(position).title);
        return v;*/
    }

}
